backend/corpora/lambdas/api/v1/collection_uuid/upload.py

```diff
@@ -17,22 +20,32 @@
     NonExistentCollectionException,
     InvalidProcessingStateException,
     NonExistentDatasetException,
+    DuplicateTagExistsException,
 )
 
 
 def link(collection_uuid: str, body: dict, token_info: dict):
-    dataset_id = upload_from_link(collection_uuid, token_info, body["url"])
+    dataset_id = upload_from_link(collection_uuid, token_info, body["url"], curator_tag=body.get("curator_tag"))
     return make_response({"dataset_uuid": dataset_id}, 202)
 
 
 def relink(collection_uuid: str, body: dict, token_info: dict):
-    dataset_id = upload_from_link(collection_uuid, token_info, body["url"], body["id"])
+    dataset_id = upload_from_link(
+        collection_uuid,
+        token_info,
+        body.get("url", body.get("link")),
+        body.get("id"),
+        curator_tag=body.get("curator_tag"),
+    )
     return make_response({"dataset_uuid": dataset_id}, 202)
 
 
 @dbconnect
-def upload_from_link(collection_uuid: str, token_info: dict, url: str, dataset_id: str = None):
+def upload_from_link(collection_uuid: str, token_info: dict, url: str, dataset_id: str = None, curator_tag: str = None):
     db_session = g.db_session
+    get_collection(
+        db_session, collection_uuid, visibility=CollectionVisibility.PRIVATE.name, owner=owner_or_allowed(token_info)
+    )
```

Comment:
So we're using this to check ownership? I think we should add a comment here, otherwise it seems like this statement has no effect.